### PR TITLE
DNM: oracle hooks use symbol denom

### PR DESCRIPTION
**Pull request by toteki**
Case-insensitive symbol denom matching is used instead of base denom matching.

Also, when the leverage registry updates a base denom or exponent, the oracle accept list also updates with it (unless the token is blacklisted)

### CODE REVIEW

File: x/oracle/keeper/hooks.go [hunk 1]
```diff
@@ -33,8 +35,12 @@
 
 	var tokenExists bool
 	for _, t := range acceptList {
-		if t.BaseDenom == token.BaseDenom {
+		// On case-insensitive match of symbol denom, update base denom and exponent
+		// to match the leverage registry. Does not modify existing symbol denom.
+		if strings.EqualFold(t.SymbolDenom, token.SymbolDenom) {
 			tokenExists = true
+			t.BaseDenom = token.BaseDenom
```
[the remark this answers is not in the document]
Comment (toteki) [reply x/oracle/keeper/hooks.go]: Switched `t` to `acceptList[i]` to solve

File: x/oracle/keeper/hooks.go [hunk 2]
```diff
@@ -32,9 +34,13 @@ func (h Hooks) AfterTokenRegistered(ctx sdk.Context, token leveragetypes.Token)
 	acceptList := h.k.AcceptList(ctx)
 
 	var tokenExists bool
-	for _, t := range acceptList {
-		if t.BaseDenom == token.BaseDenom {
+	for i, t := range acceptList {
+		// On case-insensitive match of symbol denom, update base denom and exponent
+		// to match the leverage registry. Does not modify existing symbol denom.
+		if strings.EqualFold(t.SymbolDenom, token.SymbolDenom) {
```
Comment (robert-zaremba) [inline x/oracle/keeper/hooks.go:40]: could you add a test for this fix?

**Review by robert-zaremba — COMMENTED** [review]
let's add a test.

**Review by adamewozniak — COMMENTED** [review]
Not really sure about this one - especially if we come into a scenario where we want to use the same symbol for multiple assets (e.g., supporting bridgeless & grav assets at the same time)

Let's just have an enum in the leverage hooks which allows the proposer to define how it treats the relationship between the oracle params & leverage registry

**Comment by robert-zaremba** [comment]
you mean an enum to decide if we match by symbol or by denom? Maybe we should extend the type and add `MatchDenom` field, which will be used if defined (otherwise base denom will be used)?

**Comment by adamewozniak** [comment]
@toteki wdyt about Robert's suggestion above?

**Comment by toteki** [comment]
Not sure - I'm wondering if we can't just merge this as-is and make changes if we ever need to support same-symbol assets, because duplicate accept list entries cause (voting) errors currently anyway.

**Pull request closed without merging**